apps/api/src/app/integrations/dtos/integration-response.dto.ts

```diff
@@ -45,4 +47,10 @@ export class IntegrationResponseDto {
 
   @ApiProperty()
   primary: boolean;
+
+  @ApiPropertyOptional({
+    type: [StepFilter],
+  })
+  @ValidateNested({ each: true })
```

Comment:
on the response, we don't need validation ;) probably you mixed it up with request dto: `CreateIntegrationRequestDto`